src/main/java/com/owncloud/android/lib/resources/activities/GetActivitiesRemoteOperation.java

```diff
@@ -180,17 +181,27 @@ public boolean hasMoreActivities() {
     }
 
     protected ArrayList<Activity> parseResult(String response) {
-        JsonParser jsonParser = new JsonParser();
-        JsonObject jo = (JsonObject)jsonParser.parse(response);
-        JsonArray jsonDataArray = jo.getAsJsonObject(NODE_OCS).getAsJsonArray(NODE_DATA);
+        if (response == null || response.isEmpty()) {
+            return new ArrayList<>();
+        }
+
+        try {
+            JsonParser jsonParser = new JsonParser();
+            JsonObject jo = (JsonObject) jsonParser.parse(response);
+            JsonArray jsonDataArray = jo.getAsJsonObject(NODE_OCS).getAsJsonArray(NODE_DATA);
 
-        Gson gson = new GsonBuilder()
-                .registerTypeAdapter(RichElement.class, new RichElementTypeAdapter())
-                .registerTypeAdapter(PreviewObject.class, new PreviewObjectAdapter())
-                .create();
-        Type listType = new TypeToken<List<Activity>>(){}.getType();
+            Gson gson = new GsonBuilder()
+                    .registerTypeAdapter(RichElement.class, new RichElementTypeAdapter())
+                    .registerTypeAdapter(PreviewObject.class, new PreviewObjectAdapter())
+                    .create();
+            Type listType = new TypeToken<List<Activity>>() {
+            }.getType();
 
-        return gson.fromJson(jsonDataArray, listType);
+            return gson.fromJson(jsonDataArray, listType);
+
+        } catch (JsonSyntaxException e) {
+            return new ArrayList<>();
```

Comment:
Should at least have a log statement so we can see what happened instead of just geeting an empty result (like in the first line...)

Reply:
Good idea.
I tried it a moment ago, but then the test fails, as Log_OC is not mocked.
I have a look.

Reply:
Done